sim-cli/src/main.rs

```diff
@@ -199,3 +217,47 @@ async fn main() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+async fn read_sim_path(data_dir: PathBuf, sim_file: PathBuf) -> anyhow::Result<PathBuf> {
+    let sim_path = if sim_file.is_relative() {
+        data_dir.join(sim_file)
+    } else {
+        sim_file
+    };
+
+    if sim_path.exists() {
+        Ok(sim_path)
+    } else {
+        log::info!("Simulation file '{}' does not exist", sim_path.display());
+        let sim_files: Vec<String> = std::fs::read_dir(data_dir.clone())?
```

Comment:
Either `: Vec<String>` or `::Vec<_>>()` (when collecting) is redundant. One of them can be dropped